libraries/common/subscriptions/router.js

```diff
@@ -64,10 +69,30 @@ export default function router(subscribe) {
       }
     }
 
-    // Check for a redirect url and change location if one is found.
-    const redirect = redirects.get(location);
+    // Check for a redirect and change location if one is found.
+    let redirect = redirects.get(location);
 
     if (redirect) {
+      // Check if a redirect handler was assigned and execute it.
+      if (typeof redirect === 'function' || redirect instanceof Promise) {
+        const { pathname } = getCurrentRoute();
+        dispatch(setViewLoading(pathname));
+
+        try {
+          redirect = await redirect(params);
+        } catch (e) {
+          redirect = null;
+          logger.error(e);
+        }
+
+        dispatch(unsetViewLoading(pathname));
+
+        if (!redirect) {
+          // Stop processing when no redirect was created.
```

Comment:
Also obvious. The code speaks for itself.

Reply:
removed